compliant/src/lib.rs

```diff
@@ -0,0 +1,1171 @@
+            compute_framework: parts[2].to_string(),
+            arch: parts[3].to_string(),
+            os: parts[4].to_string(),
+        })
+    }
+}
+
+/// Struct to hold repository list result
+#[derive(Serialize)]
+pub struct RepoListResult {
+    pub repositories: Vec<String>,
+    pub count: usize,
+}
+
+/// Struct for console output formatting
+pub struct ConsoleFormatter;
+
+impl ConsoleFormatter {
+    pub fn format_repo_list(repos: &[String], count: usize) {
+        println!(".");
+        for repo_id in repos {
+            println!("├── {}", repo_id);
+        }
+        println!("╰── {} kernel repositories found\n", count);
+    }
+
+    pub fn format_missing_repo(repo_id: &str) {
+        println!(".");
+        println!("├── {}", repo_id.on_bright_white().black().bold());
+        println!("├── build: missing");
+        println!("╰── abi: missing");
+    }
+
+    pub fn format_fetch_status(repo_id: &str, fetching: bool, result: Option<&str>) {
+        println!("repository: {}", repo_id);
+        if fetching {
+            println!("status: not found locally, fetching...");
+        }
+        if let Some(message) = result {
+            println!("status: {}", message);
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn format_repository_check_result(
+        repo_id: &str,
+        build_status: &str,
+        cuda_compatible: bool,
+        #[cfg(feature = "enable_rocm")] rocm_compatible: bool,
+        #[cfg(not(feature = "enable_rocm"))] _rocm_compatible: bool,
+        cuda_variants: &[String],
+        #[cfg(feature = "enable_rocm")] rocm_variants: &[String],
+        #[cfg(not(feature = "enable_rocm"))] _rocm_variants: &[String],
+        cuda_variants_present: Vec<&String>,
+        #[cfg(feature = "enable_rocm")] rocm_variants_present: Vec<&String>,
+        #[cfg(not(feature = "enable_rocm"))] _rocm_variants_present: Vec<&String>,
+        compact_output: bool,
+        abi_output: &AbiCheckResult,
+        abi_status: &str,
+    ) {
+        // Display console-formatted output
+        let abi_mark = if abi_output.overall_compatible {
+            "✓".green()
+        } else {
+            "✗".red()
+        };
+
+        let cuda_mark = if cuda_compatible {
+            "✓".green()
+        } else {
+            "✗".red()
+        };
+
+        #[cfg(feature = "enable_rocm")]
+        let rocm_mark = if rocm_compatible {
+            "✓".green()
+        } else {
+            "✗".red()
+        };
+
+        let label = format!(" {} ", repo_id).black().on_bright_white().bold();
+
+        println!("\n{}", label);
+        println!("├── build: {}", build_status);
+
+        if !compact_output {
+            println!("│  {} {}", cuda_mark, "CUDA".bold());
+
+            // Print variant list with proper tree characters
+            for (i, cuda_variant) in cuda_variants.iter().enumerate() {
+                let is_last = i == cuda_variants.len() - 1;
```

Comment:
I think using `Peekable` is nicer/less fragile. Something like:

```suggestion
            let iter = cuda_variants.iter().peekable();
            while let Some(cuda_variant) = iter.nex() {
                let is_last = iter.peek().is_none();
```

Reply:
oh this is way nicer, thanks for the suggestion! updated in the latest commits